IDR/notebook/scripts/nbconverted/1.describe_screens.py

```diff
@@ -140,15 +165,65 @@ def describe_screen(screen_id):
                 # Clean channels value and separate into stain and stain target
                 stain = set()
                 stain_target = set()
-                for entry in channels.split(";"):
-                    temp_list = entry.split(":")
-                    stain.add(temp_list[0])
-                    stain_target.add(temp_list[1])
+                for channel in channels.split(";"):
+                    # For channel entries with 'stain:target' format
+                    if bool(re.search(r"([\:])+", channel)):
+                        # Separate stain and target
+                        st_list = channel.split(":")
+                        # Add to respective sets
+                        stain.add(st_list[0])
+                        stain_target.add(st_list[1])
+
+                    # For channel entries with 'stain (target)' format
+                    elif bool(channel[channel.find("(") + 1 : channel.rfind(")")]):
+                        # Search for target name within parentheses
+                        target_name = channel[
+                            channel.find("(") + 1 : channel.rfind(")")
+                        ]
+                        # Remove parentheses for stain name
+                        stain_name = re.sub(r"\([^()]*\)", "", channel)
+                        # Remove redundancies (ie. if (target (abbreviation)) present)
+                        processed = {"target": target_name, "stain": stain_name}
```

Comment:
can you use a bit more descriptive of a variable name other than `processed`? What are you processing?

Reply:
The dictionary name was changed to `attributes` instead. The `processed` name was just a placeholder I forgot to change.

Added in 5496254ddce2e5b9ac546300eb4c67471deac700.

Reply:
`attributes` is still kinda weak - what kind of attributes? What attributes are you processing?

How about `image_attributes`?

Reply:
I see how `attributes` could be too general. I've only been thinking about attributes only as image attributes for a month straight now so it made sense to me. The dictionary name was changed to `image_attributes`.

Added in c7cac9b5eadaef2091a706dd97ab752e1f1afe63.